src/dmd/template.h

```diff
@@ -273,6 +271,46 @@ class TemplateInstance : public ScopeDsymbol
     TemplateInstance *tnext;            // non-first instantiated instances
     Module *minst;                      // the top module that instantiated this instance
 
+private:
+    unsigned short _nest;                // for recursive pretty printing detection, 3 MSBs reserved for flags (below)
+public:
+    unsigned char inuse;                 // for recursive expansion detection
+
+private:
```

Comment:
Don't need have this enum or accessor functions exposed to C++. Just mark them extern(D) internally.

Reply:
So you mean just remove them from `template.h`?

Reply:
Yeah. The more that's kept internal, the better.

Reply:
done, please take a look